receiver/mongodbatlasreceiver/logs.go

```diff
@@ -0,0 +1,311 @@
+	consumer    consumer.Logs
+	stopperChan chan struct{}
+	wg          sync.WaitGroup
+}
+
+type resourceInfo struct {
+	Org      string
+	Project  mongodbatlas.Project
+	Cluster  mongodbatlas.Cluster
+	Hostname string
+	LogName  string
+	start    time.Time
+	end      time.Time
+}
+
+// MongoDB Atlas Documentation reccommends a polling interval of 5  minutes: https://www.mongodb.com/docs/atlas/reference/api/logs/#logs
+const collectionInterval = time.Minute * 5
+
+func newMongoDBAtlasLogsReceiver(settings component.ReceiverCreateSettings, cfg *Config, consumer consumer.Logs) (*logsReceiver, error) {
+	client, err := internal.NewMongoDBAtlasClient(cfg.PublicKey, cfg.PrivateKey, cfg.RetrySettings, settings.Logger)
+	if err != nil {
+		return nil, err
+	}
+	recv := &logsReceiver{log: settings.Logger, cfg: cfg, client: client, stopperChan: make(chan struct{}), consumer: consumer}
+	return recv, nil
+}
+
+// Log receiver logic
+func (s *logsReceiver) Start(ctx context.Context, host component.Host) error {
+	s.wg.Add(1)
+	go func() {
+		defer s.wg.Done()
+		s.collect(ctx)
+	}()
+	return nil
+}
+
+func (s *logsReceiver) Shutdown(ctx context.Context) error {
+	close(s.stopperChan)
+	s.wg.Wait()
+	return s.shutdown(ctx)
+}
+
+func (s *logsReceiver) shutdown(context.Context) error {
+	return s.client.Shutdown()
+}
+
+// parseHostName parses out the hostname from the specified cluster host
+func parseHostName(s string, logger *zap.Logger) []string {
+	var hostnames []string
+
+	// first check to make sure string is not empty
+	if s == "" {
+		return []string{}
+	}
+
+	// create an array with a comma delimiter from the original string
+	tmp := strings.Split(s, ",")
+	for _, t := range tmp {
+
+		h := strings.TrimPrefix(t, "mongodb://")
+
+		// separate hostname from scheme and port
+		host, _, err := net.SplitHostPort(h)
+		if err != nil {
+			logger.Error("Could not parse out hostname: " + host)
+			continue
+		}
+		// hostname parsed successfully
+		hostnames = append(hostnames, host)
+	}
+
+	return hostnames
+}
+
+// collect spins off functionality of the receiver from the Start function
+func (s *logsReceiver) collect(ctx context.Context) {
+	resource := resourceInfo{start: time.Now()}
+	for {
+		// collection interval loop,
+		select {
+		case <-ctx.Done():
+			return
+		case <-s.stopperChan:
+			return
+		case <-time.After(collectionInterval):
+		}
+
+		for _, p := range s.cfg.Logs.Projects {
+			project, err := s.client.GetProject(ctx, p.Name)
+			if err != nil {
+				s.log.Error("Error retrieving project "+p.Name+":", zap.Error(err))
+				continue
+			}
+
+			org, err := s.client.GetOrganization(ctx, project.OrgID)
+			if err != nil {
+				s.log.Error("Error retrieving organization", zap.Error(err))
+				resource.Org = "unknown"
+			} else {
+				resource.Org = org.Name
+			}
+
+			// get clusters for each of the projects
+			resource.Project = *project
+			clusters, err := s.processClusters(ctx, *p, &resource)
```

Comment:
This is a good example of where variable naming needs to be cleaned up. At this point, we have a variables `project` and `p`. We pass in `p` to a function that then calls it `project`. This is difficult to follow.